src/main/flight/pid.c

```diff
@@ -277,7 +277,24 @@ void pidResetIterm(void)
 void pidUpdateAntiGravityThrottleFilter(float throttle)
 {
     if (pidRuntime.antiGravityMode == ANTI_GRAVITY_SMOOTH) {
-        pidRuntime.antiGravityThrottleHpf = throttle - pt1FilterApply(&pidRuntime.antiGravityThrottleLpf, throttle);
+        // calculate a boost factor for P in the same way as for I when throttle changes quickly
+        // at this point the variable antiGravityThrottleHpf is the lowpass of throttle
+        pidRuntime.antiGravityThrottleHpf = pt1FilterApply(&pidRuntime.antiGravityThrottleLpf, throttle);
+        // focus P boost on low throttle range only
+        if (throttle < 0.5f) {
+            pidRuntime.antiGravityPBoost = 0.5f - throttle;
+        } else {
+            pidRuntime.antiGravityPBoost = 0.0f;
+        }
+        // use lowpass to identify start of a throttle up, use this to reduce boost at start by half
+        if (pidRuntime.antiGravityThrottleHpf < throttle) {
+            pidRuntime.antiGravityPBoost *= 0.5f;
+        }
```

Comment:
This condition will be alway true even if the throttle stopped after a very slow rise.
To make this condition true only when throttle rise can we imagine something like  ? 

   if (pidRuntime.antiGravityThrottleHpf < throttle*0.95f) {
            pidRuntime.antiGravityPBoost *= 0.5f;
   }

Reply:
Yes, that code is true whenever throttle is changed suddenly after a period of stable throttle.

However it is good like this, because the same errors exist every time we suddenly change throttle, regardless of where we start, though mostly only at the low end of the throttle.

Race pilots have noticed improved stability when they transiently chop and add throttle, so I think it is good the way it is.

Reply:
sorry but : "This condition will be alway true even if the throttle stopped after a very slow rise."  and I suppose it is not what you want.

if you have a value going from 0 to 1 and you apply a lpf on it the value will NEVER reach 1. the value will be 0,9 -> 0,99 -> 0,999 -> 0,9999 .... (regardless of the speed of change from 0 to 1).

Reply:
Ah I see what you mean.
In practice, however, the code won't have a problem the way it is, and the multiplication by 0.95 won't be needed, for two reasons:
- the code does nothing above throttle of 0.5, and
- the code does nothing if the sticks are still, or barely moving.
If he throttle has been pushed above 0.5 to max and held still, there will be no problem with the condition being true since there will be no boost above 0.5 throttle
If the stick is pushed say to 0.4 and somehow held perfectly still, then there will be no boost to apply, so it won't matter if this detector toggles on or off, there is nothing to add.
The only purpose of this detector is to reduce the boost applied when sticks are rising compared to when they are falling.  It doesn't matter if they are still.
If there is a way to simplify that aspect of the coding - to reduce the number of conditionals required - that would be great, but this suggestion requires a multiplication that isn't needed.